tests/monitors/kubernetes_cluster/kubernetes_cluster_test.py

```diff
@@ -205,3 +205,22 @@ def test_cronjobs(k8s_cluster):
                 ),
                 timeout_seconds=300,
             )
+
+
+@pytest.mark.kubernetes
+def test_pods(k8s_cluster):
+    config = """
+    monitors:
+     - type: kubernetes-cluster
+    """
+    yamls = [SCRIPT_DIR / "resource_quota.yaml", TEST_SERVICES_DIR / "nginx/nginx-k8s.yaml"]
+    with k8s_cluster.create_resources(yamls):
+        with k8s_cluster.run_agent(agent_yaml=config) as agent:
+            assert wait_for(
```

Comment:
Also, the `k8s_cluster.create_resources` call will yield the created resources, so that might help too.

Reply:
Updated tests to look for specific values. Since `create_resources` does not return nested . workloads, I had to make use of the API to get pods from a deployment.

Reply:
All you have to do is query for pods in the same namespace (`k8s_cluster.test_namespace` gives you the namespace here) by looking up the label.  That is how Kubernetes itself manages the link between Deployment and Pods.  See https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#list_namespaced_pod and the `label_selector` kwarg.

Reply:
Updated this to just use `list_namespaced_pod` instead